typescript/cli/src/context/context.ts

```diff
@@ -199,3 +203,35 @@ function getSubmissionStrategy(
   );
   return SubmissionStrategySchema.parse(submissionStrategyFileContent);
 }
+
+export async function getOrRequestApiKeys(
+  chains: ChainName[],
+  chainMetadata: ChainMap<ChainMetadata>,
+): Promise<ChainMap<string>> {
+  const apiKeys: ChainMap<string> = {};
+
+  for (const chain of chains) {
+    const wantApiKey = await confirm({
+      default: false,
+      message: `Do you want to use an API key to verify on this (${chain}) chain's block explorer`,
+    });
+    if (wantApiKey) {
+      apiKeys[chain] = await detectAndConfirmOrPrompt(
+        async () => {
+          const blockExplorers = chainMetadata[chain].blockExplorers;
+          if (!(blockExplorers && blockExplorers.length > 0)) return;
+          for (const blockExplorer of blockExplorers) {
+            /* The current apiKeys mapping only accepts one key, even if there are multiple explorer options present. */
```

Comment:
related issue: https://github.com/hyperlane-xyz/issues/issues/1105

would add AC for that ticket to refactor apikey management across the board so we could have a key per explorer as opposed to key per chain